### PR TITLE
std::array removals

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

**Comment by neheb** [comment]
@kmilos should fix CLANG64 compilation.

**Comment by neheb** [comment]
From my reading of the code, Exiv2::ExifData::const_iterator evaluates to std::list<Exifdatum>::const_iterator. It's interesting that only CLANG64 on MSYS2 fails to compile.

**Comment by kmilos** [comment]
> Exiv2::ExifData::const_iterator evaluates to std::list::const_iterator

Yeah, but we're not initializing the array with the return value, but with the function pointer, no? And that's not a literal either, but should be known/constant at compile time?

Anyhow, it would good if @1div0 could spare us a minute and try main w/ -DBUILD_SAMPLES=ON on his F38 using Clang 15 as well please?

File: src/cr2image.cpp [hunk 1]
```diff
@@ -109,10 +109,7 @@ WriteMethod Cr2Parser::encode(BasicIo& io, const byte* pData, size_t size, ByteO
   ExifData ed = exifData;
 
   // Delete IFDs which do not occur in TIFF images
-  static constexpr auto filteredIfds = std::array{
-      IfdId::panaRawId,
-  };
-  for (auto&& filteredIfd : filteredIfds) {
+  for (auto&& filteredIfd : {IfdId::panaRawId}) {
```
Comment (kmilos) [inline src/cr2image.cpp:112]: I'd maybe leave all of these 1-element arrays as is, how can we be sure we won't be adding more elements in the future?

**Comment by 1div0** [comment]
> > Exiv2::ExifData::const_iterator evaluates to std::list::const_iterator
> 
> Yeah, but we're not initializing the array with the return value, but with the function pointer, no? And that's not a literal either, but should be known/constant at compile time?
> 
> Anyhow, it would good if @1div0 could spare us a minute and try main w/ -DBUILD_SAMPLES=ON on his F38 using Clang 15 as well please?

Sure, will do in the evening or WE. 2 busy now.

File: src/cr2image.cpp (hunk 1 above)
Comment (neheb) [reply src/cr2image.cpp:112]: Will do

**Review by kmilos — APPROVED** [review]
LGTM, but let's not merge just yet, until @1div0 can give feedback on main as is

**Comment by neheb** [comment]
OK

**Comment by 1div0** [comment]
[Exiv2 LLVM 15.0.7 F38 samples.log](https://github.com/Exiv2/exiv2/files/11055722/Exiv2.LLVM.15.0.7.F38.samples.log)

**Comment by neheb** [comment]
That error seems unrelated. Also, why are you building Jzon.cpp? That was removed from CMake quite a while ago.

**Pull request merged**